synapse/util/caches/lrucache.py

```diff
@@ -154,14 +154,14 @@ def cache_get(key, default=None, callbacks=[]):
         def cache_set(key, value, callbacks=[]):
             node = cache.get(key, None)
             if node is not None:
-                if value != node.value:
+                if node.callbacks and value != node.value:
```

Comment:
Oh, this was just a cheeky thing to only do the `!=` check if we do actually have `node.callbacks`